internal/luks2/fifo.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package luks2
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+	"golang.org/x/xerrors"
+)
+
+func mkFifo() (string, func(), error) {
+	// /run is not world writable but we create a unique directory here because this
+	// code can be invoked by a public API and we shouldn't fail if more than one
+	// process reaches here at the same time.
+	dir, err := ioutil.TempDir(runDir, filepath.Base(os.Args[0])+".")
+	if err != nil {
+		return "", nil, xerrors.Errorf("cannot create temporary directory: %w", err)
+	}
+
+	cleanup := func() {
+		os.RemoveAll(dir)
```

Comment:
BTW. should we maybe log something to stderr if this fails?

Reply:
I've updated this to log a message to stderr if it fails (and in a few other places too).